ProjectCoimbra.UWP/Project.Coimbra.Midi/MidiEngine.cs

```diff
@@ -270,14 +305,16 @@ private void AddInstruments(MidiFile midi)
 
                 if (this.Instruments.ContainsKey(currentEvent.Channel))
                 {
-                    if (!this.Instruments[currentEvent.Channel].Contains(programNumber))
+                    if (!this.Instruments[currentEvent.Channel].ProgramNumbers.Contains(programNumber))
```

Comment:
You might be able to use [`TryAdd`](https://docs.microsoft.com/dotnet/api/system.collections.generic.dictionary-2.tryadd) here although I'm not sure it's available for the version of .NET we're using.